From 902494619367416a10ecb5c0e6dcc60941142c18 Mon Sep 17 00:00:00 2001
From: Randy Li <ayaka@soulik.info>
Date: Mon, 2 Nov 2020 08:46:25 +0000
Subject: [PATCH 42/60] wlvideoformat: fix DMA format convertor

In the most of case, this typo would work. But for
ARGB8888 and XRGB8888, which shm format is not based on fourcc,
which would never appear in format enumeration.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/1751>
(cherry picked from commit 4e4c26af4be53088d28b37017a7554299dae1aa6)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 ext/wayland/wlvideoformat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/wayland/wlvideoformat.c b/ext/wayland/wlvideoformat.c
index e1d6271ea..68cec5093 100644
--- a/ext/wayland/wlvideoformat.c
+++ b/ext/wayland/wlvideoformat.c
@@ -112,7 +112,7 @@ gst_wl_dmabuf_format_to_video_format (guint wl_format)
   guint i;
 
   for (i = 0; i < G_N_ELEMENTS (wl_formats); i++)
-    if (wl_formats[i].wl_shm_format == wl_format)
+    if (wl_formats[i].dma_format == wl_format)
       return wl_formats[i].gst_format;
 
   return GST_VIDEO_FORMAT_UNKNOWN;
-- 
2.20.1

